ion/cmp_test.go

```diff
@@ -211,7 +246,7 @@ func containersEquality(this, other interface{}) bool {
 	default:
 		otherItem := other.(ionItem)
 		thisItem := this.(ionItem)
-		if thisItem.fieldName == otherItem.fieldName && thisItem.equal(otherItem) {
+		if thisItem.fieldName.Equal(&otherItem.fieldName) && thisItem.equal(otherItem) {
```

Comment:
Currently equal() function takes in a non-pointer
`func (i *ionItem) equal(o ionItem) bool {`

If I were to do your suggestion it would be:
```
	default:
		otherItem := other.(*ionItem)
		thisItem := this.(*ionItem)
		if thisItem.fieldName.Equal(&otherItem.fieldName) && thisItem.equal(*otherItem) {
			return true
		}
```

but it is failing right now because the value APIs arent a pointer yet